src/components/switchComponent/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC, ReactNode } from 'react';
+import {
+  Switch as ChakraSwitch,
+  SwitchCheckedChangeDetails,
+  SystemStyleObject,
+  useSlotRecipe,
+} from '@chakra-ui/react';
+
+export type SwitchComponentProps = {
+  onChange: (details: SwitchCheckedChangeDetails) => void;
+  isDisabled: boolean;
+  isChecked: boolean;
```

Comment:
Chakra 3 renamed boolean and removed `is`, let's follow that pattern for consistency
```suggestion
  disabled: boolean;
  checked: boolean;
```